1. 

(X_train=80%)
(X_test=20%)

A partir del archivo "iris.csv" que se corresponde con datos relacionados con tres tipos de plantas (el target es el tipo de planta):

- Mediante regresión logística, predice qué planta se corresponde con los ejemplos: [[15.7,2.8,9.5,0.1]] y [[1.7,3.8,4.5,11.3]]
- ¿Qué acierto tiene tu modelo? (score)
- Utiliza un conjunto de test con el 40% de los datos: ¿cambian los resultados anteriores? ¿por qué?
- Elimina la clase "Virginica" de tus datos. Repite el proceso anterior, ¿cambian los resultados?
- ¿Consideras la regresión logística un buen algoritmo para estos datos? ¿por qué?

In [1]:
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt
import seaborn as sns
sns.set_style('whitegrid')
%matplotlib inline


In [2]:
iris_df = pd.read_csv('../data/iris.csv')

In [3]:
X =  iris_df[['sepal.length', 'sepal.width', 'petal.length', 'petal.width']]
y = iris_df['variety']

In [4]:
from sklearn.model_selection import train_test_split
# Split dataset into training set and test set
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.20)
print(X_train.shape)
print(X_test.shape)

(120, 4)
(30, 4)


In [5]:
from sklearn.linear_model import LogisticRegression
model = LogisticRegression()
model.fit(X_train, y_train)

LogisticRegression()

### Predice qué planta se corresponde con los ejemplos: [15.7,2.8,9.5,0.1] y [1.7,3.8,4.5,11.3]

In [6]:
predicted = model.predict([[15.7,2.8,9.5,0.1]]) 
print(predicted)

['Virginica']


In [7]:
predicted = model.predict([[1.7,3.8,4.5,11.3]]) 
print(predicted)

['Virginica']


In [8]:
y_pred = model.predict(X_test)
y_pred

array(['Virginica', 'Virginica', 'Versicolor', 'Versicolor', 'Virginica',
       'Versicolor', 'Versicolor', 'Virginica', 'Virginica', 'Virginica',
       'Virginica', 'Setosa', 'Versicolor', 'Setosa', 'Versicolor',
       'Virginica', 'Versicolor', 'Setosa', 'Versicolor', 'Virginica',
       'Setosa', 'Virginica', 'Virginica', 'Versicolor', 'Virginica',
       'Virginica', 'Setosa', 'Setosa', 'Setosa', 'Setosa'], dtype=object)

### ¿Qué acierto tiene tu modelo? (score)

In [9]:
#Import scikit-learn metrics module for accuracy calculation
from sklearn import metrics
# Model Accuracy, how often is the classifier correct?
print("Accuracy:",metrics.accuracy_score(y_test, y_pred))

Accuracy: 0.9333333333333333


### Cambiando el conjunto de test al 40%

In [10]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.40)

In [11]:
model = LogisticRegression()
model.fit(X_train, y_train)

LogisticRegression()

In [12]:
y_pred = model.predict(X_test)

In [13]:
print("Accuracy:",metrics.accuracy_score(y_test, y_pred))

Accuracy: 0.9833333333333333


### Eliminando la clase Virginica

In [14]:
iris_df = iris_df[iris_df.variety !='Virginica']

In [15]:
iris_df.variety.value_counts()

Setosa        50
Versicolor    50
Name: variety, dtype: int64

In [16]:
X =  iris_df[['sepal.length', 'sepal.width', 'petal.length', 'petal.width']]
y = iris_df['variety']

In [17]:
X.shape

(100, 4)

In [18]:
y.shape

(100,)

In [19]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.20)

In [20]:
model = LogisticRegression()
model.fit(X_train, y_train)

LogisticRegression()

In [21]:
y_pred = model.predict(X_test)

In [22]:
print("Accuracy:",metrics.accuracy_score(y_test, y_pred))

Accuracy: 1.0


### ¿Consideras la regresión logística un buen algoritmo para estos datos? ¿por qué?
`Si. Viendo el porcentaje de acierto conseguido (mayor al 93%) considero que es un buen algoritmo`